crates/compiler/gen_llvm/src/llvm/lowlevel.rs

```diff
@@ -608,6 +608,28 @@ pub(crate) fn run_low_level<'a, 'ctx>(
                 bitcode::STR_WITH_CAPACITY,
             )
         }
+        StrConcatUtf8 => {
+            // Str.concatUtf8: List U8, Str -> List U8
+            arguments!(list, string);
+
+            // XXX: I don't know why I need to call different functions based on the target pointer width, but the test_gen tests panic if I don't
+            match env.target.ptr_width() {
```

Comment:
I could use some help explaining why this is necessary, because I didn't understand the comment on `Str.concat`, I think because I don't know why I would use `call_str_bitcode_fn` vs `call_list_bitcode_fn` and I'm don't understand why the string be a pointer or not on different architectures affects the decision to of which `call_*_bitcode_fn` to use.

Reply:
the logic is just subtly different between 32-bit and 64-bit platforms. It turns out that the list logic also works for strings on 64-bit systems.

Reply:
Should I remove the branch and just call the `call_list_bitcode_fn` or keep the branch?